### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by IvanGuculyak**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,34 +1,55 @@
...
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
+        self.is_drowned = is_drowned
+        self.decks = []
+        if start == end:
+            self.decks.append(Deck(start[0], start[1]))
+        elif start[0] == end[0] and start[1] != end[1]:
+            for i in range(start[1], end[1] + 1):
```
Comment (vasylhnatiuk) [inline app/main.py]: `i` is a bad name for variable

File: app/main.py [hunk 2]
```diff
@@ -1,34 +1,55 @@
...
+                self.decks.append(Deck(start[0], i))
+        elif start[0] != end[0] and start[1] == end[1]:
+            for i in range(start[0], end[0] + 1):
```
Comment (vasylhnatiuk) [inline app/main.py]: the same here

File: app/main.py [hunk 3]
```diff
@@ -1,34 +1,55 @@
...
+            for deck in battle_ship.decks:
+                self.field[(deck.row, deck.column)] = battle_ship
 
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def fire(self, location: tuple) -> str:
```
Comment (vasylhnatiuk) [inline app/main.py]:
```suggestion
    def fire(self, location: tuple[int]) -> str:
```

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 4]
```diff
@@ -1,34 +1,55 @@
...
+        self.is_drowned = is_drowned
+        self.decks = []
+        if start == end:
+            self.decks.append(Deck(start[0], start[1]))
+        elif start[0] == end[0] and start[1] != end[1]:
+            for column in range(start[1], end[1] + 1):
+                self.decks.append(Deck(start[0], column))
+        elif start[0] != end[0] and start[1] == end[1]:
+            for row in range(start[0], end[0] + 1):
+                self.decks.append(Deck(row, start[1]))
```
Comment (vsmutok) [inline app/main.py]: put it in a separate method

**Review by vsmutok — CHANGES_REQUESTED** [review]

**Review by vsmutok — APPROVED** [review]
Good Job 👍

**Pull request closed without merging**